core/iwasm/aot/aot_intrinsic.c

```diff
@@ -474,6 +481,32 @@ aot_intrinsic_f64_cmp(AOTFloatCond cond, float64 lhs, float64 rhs)
     return 0;
 }
 
+int32
+aot_intrinsic_cmpxchg_4(int32 *ptr, int32 expected, int32 desired, int success,
+                        int failure)
+{
+    int32 ret = *ptr;
+#if !defined(BUILD_TARGET_X86_64) && !defined(BUILD_TARGET_X86_32) \
+    && !defined(BUILD_TARGET_AMD64)
+    __atomic_compare_exchange_4(ptr, &expected, desired, false, success,
```

Comment:
Can we make sure that __atomic_compare_exchange_4 is supported by most compilers? Not sure whether it might cause link error.

Reply:
How about only enable it on tested platform and raise a runtime log on others ?

Reply:
Not known why we need to register aot_intrinsic_cmpxchg_4/8 global symbol map in aot_reloc.h? It seems that only riscv needs these APIs in multi-thread mode, shall we move them into aot_reloc_riscv.c?

Reply:
Use LLVM builtin pass to convert these instructions